File: irrgarten/src/irrgarten/GameState.java
package irrgarten;

/**
<<<<<<< HEAD
 *
=======
 * @author Juan Molina Molina
 * @author Paula Jiménez Ruiz
>>>>>>> 1fe4584be61467713d9bee7c1973916b27fcf79a
 */
public class GameState {

    
    private String labyrinth;
    private String players;
    private String monsters;
    private int currentPlayer;
    private boolean winner;
    private String log;
    
    public GameState(String l, String p, String m, int cP, boolean w, String lo) {
    
        labyrinth=l;
        players=p;
        monsters=m;
        currentPlayer=cP;
        winner=w;
        log= lo;
    }
    
    public String getLabyrinth() {
    
        return labyrinth;
    }
    
    public String getPlayers() {
    
        return players;
    }
    
    public String getMonsters() {
    
        return monsters;
    }
    
    public int getCurrentPlayer() {
    
        return currentPlayer;
    }
    
    public boolean isWinner() {
    
        return winner;
    }
    
    public String getLog() {
    
        return log;
    }

}

File: ssh/GameState.rb
#encoding:utf-8

module Irrgarten

	class GameState
		
		public 
			def initialize (l, p, m, cP, w, lo)
				@labyrinth =l
				@players =p
				@monsters =m
				@current_player= cP
				@winner =w
				@log =lo
			end
			
			attr_reader :labyrinth, :players, :monsters, :current_player, :winner, :log #Sirve para cerear instancias que se puedan leer pero no acceder desde fuera de la clase
			
			def get_labyrinth
				return @labyrinth
			end
			
			def get_players
				return @players
			end
			
			def get_monsters
				return @monsters
			end
			
			def get_current_player
				return @current_player
			end
			
			def get_winner
				return @winner
			end
			
			def get_log
				return @log
			end
	end
end
		
